google/cloud/internal/source_accumulators.h

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_SOURCE_ACCUMULATORS_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_SOURCE_ACCUMULATORS_H
+
+#include "google/cloud/future.h"
+#include "google/cloud/version.h"
+#include "absl/meta/type_traits.h"
+#include "absl/types/variant.h"
+#include <vector>
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace internal {
+
+template <typename Source>
+struct AccumulateAllEventsType {
+  using type = absl::variant<std::vector<typename Source::value_type>,
+                             typename Source::error_type>;
+};
```

Comment:
I'm wondering if you actually need both the struct and the alias. Could both of these be replaced with something like the following?

```
template <typename Source>
using AccumulateAllEventsType =
    absl::variant<std::vector<typename Source::value_type>,
                              typename Source::error_type>;
```

Reply:
Fixed.